components/Terms/Edit.js

```diff
@@ -12,21 +12,31 @@ export default class Edit extends Component {
 	static propTypes = {
 		term: PropTypes.Term.isRequired,
 	}
+
+	constructor(props) {
+		super(props)
+		this.state = {
```

Comment:
aren't props supposed to be readonly? 
I was following the example from http://facebook.github.io/react-native/releases/0.28/docs/textinput.html#textinput

<img width="624" alt="screenshot 2016-07-01 11 26 19" src="https://cloud.githubusercontent.com/assets/30460/16517611/a826f324-3f7e-11e6-92fd-2b6a26852ac3.png">